CRM/Core/Form.php

```diff
@@ -1249,13 +1254,22 @@ public function addField($name, $props = array(), $required = FALSE) {
         if (empty($props['multiple'])) {
           $options = array('' => $props['placeholder']) + $options;
         }
-        $this->add('select', $name, $label, $options, $required, $props);
+        $props += array('required' => $required);
+        $this->addElement('select', $name, $label, $options, $props);
```

Comment:
I don't understand this change. The `add` method enforces the required rule. AFAIK `addElement` does not.